src/com/modo/Modo2Auth.java

```diff
@@ -80,19 +80,15 @@ private String createSignature(String payload) {
         return signature.replace("+", "-").replace("/", "_").replaceAll("=+$", "");
     }
 
-    public static String toHexString(byte[] hash) {
-        // Convert byte array into signum representation
-        BigInteger number = new BigInteger(1, hash);
-
-        // Convert message digest into hex value
-        StringBuilder hexString = new StringBuilder(number.toString(16));
-
-        // Pad with leading zeros
-        while (hexString.length() < 32) {
-            hexString.insert(0, '0');
+    private static final byte[] HEX_CHARSET = "0123456789ABCDEF".getBytes(StandardCharsets.US_ASCII);
+    
+    public static String bytesToHexString(byte[] bytes) {
+        byte[] hexChars = new byte[bytes.length * 2];
+        for (int j = 0; j < bytes.length; j++) {
+            int v = bytes[j] & 0xFF;
+            hexChars[j * 2] = HEX_CHARSET[v >>> 4];
```

Comment:
I found the source on the web where this was taken, and it should work great. It's a bit opaque to read (pretty uncommon to see bit shifting) but it does the job (and fast).